prompting/validators/criteria.py

```diff
@@ -94,3 +97,72 @@ def evaluate(self, completions: List[str]) -> torch.FloatTensor:
 
     def compose_text(self) -> str:
         return self.text.format(target_length=self.target_length, unit=self.unit.value)
+    
+class LayoutTypeEnum(Enum):
+    JSON = "json"
+    DICTIONARY = "python dictionary"
+    NUMBEREDLIST = "numbered list"
+    BULLETPOINTLIST = "bullet point list"
+
+    def _select_random_attribute(self):
+        attribute_names = [attr for attr in vars(self) if not attr.startswith('_')]
+        random_attribute_name = random.choice(attribute_names)
+        random_attribute_value = getattr(self, random_attribute_name)
+        return random_attribute_value
+
+@dataclass
+class MatchLayoutCriteria(TaskCriterion):
+    text: str = "Your response must be in the form of a {format_type}{w}{fields}"
+    penalty: float = 0.1
+    format_type: LayoutTypeEnum = LayoutTypeEnum.JSON
+    num_fields: int = 0
+    fields : str = " with {num_fields} fields"
+
+    def is_json(text):
+        try:
+            json.loads(text)
+            return True
+        except ValueError:
+            return False
+        
+    def is_dictionary(text):
+        try:
+            if type(ast.literal_eval(text)) == dict:
+                return True
+            else:
+                return False
+        except (ValueError, SyntaxError):
+            return False
+
+    def is_numbered_list(input_string):
+        pattern = r'^\d.*\n?$'
+        lines = input_string.split('\n')
+        return all(re.match(pattern, line) for line in lines)
+    
+    def is_bullet_point_list(input_string):
+        pattern = r'^\s*[-*+]\s.*\n?(\s*[-*+]\s.*\n?)*$'
```

Comment:
I'd add more flexibility w.r.t. spaces since good answers may use different conventions or make use of nested lists
`r'^\s+*[-*+]\s+.*\n?(\s*[-*+]\s.*\n?)*$'`